sdk/python/feast/feature_store.py

```diff
@@ -1002,17 +991,78 @@ def get_online_features(
                 union_of_entity_keys,
             )
 
+        requested_result_row_names = self._get_requested_result_fields_and_populate_odfv_dependencies(
+            entity_name_to_join_key_map,
+            full_feature_names,
+            grouped_odfv_refs,
+            provider,
+            request_data_features,
+            needed_request_fv_features,
+            result_rows,
+            union_of_entity_keys,
+        )
+
         initial_response = OnlineResponse(
             GetOnlineFeaturesResponse(field_values=result_rows)
         )
         return self._augment_response_with_on_demand_transforms(
             _feature_refs,
-            all_on_demand_feature_views,
+            requested_result_row_names,
+            requested_on_demand_feature_views,
             full_feature_names,
             initial_response,
             result_rows,
         )
 
+    def _get_requested_result_fields_and_populate_odfv_dependencies(
+        self,
+        entity_name_to_join_key_map: Dict[str, str],
+        full_feature_names: bool,
+        grouped_odfv_refs: List[Tuple[OnDemandFeatureView, List[str]]],
+        provider: Provider,
+        request_data_features: Dict[str, List[Any]],
+        needed_request_fv_features: Set[str],
+        result_rows,
+        union_of_entity_keys,
```

Comment:
/nit I had to go and find out what these were from elsewhere in the code as I wasn't sure iterating over them would be fast - which suggests to me that having the annotations here might be useful (even if they exist elsewhere).

Reply:
Done thanks!